[ao][sparsity] Bug Fix: data norm sparsifier not working with 1D tensors/parameters

macandro96 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80465
* #80394

Issue:
Previously, the L1/L2 norm data sparsifier was not supported with
1D tensors or parameters.

Fix:
If the tensor is 1D, then unsqueeze it to make it look 2D and
perform the rest as usual. Also, added some 1D tensor in the
unit test to test this issue.

Test Plan:
```python test/test_ao_sparsity.py TestNormDataSparsifiers```

Commit:
Bug Fix: data norm sparsifier not working with 1D tensors/parameters

Issue:
Previously, the L1/L2 norm data sparsifier was not supported with
1D tensors or parameters.

Fix:
If the tensor is 1D, then unsqueeze it to make it look 2D and
perform the rest as usual. Also, added some 1D tensor in the
unit test to test this issue.

Test Plan:
```python test/test_ao_sparsity.py TestNormDataSparsifiers```

[ghstack-poisoned]

z-a-f commented on torch/ao/sparsity/experimental/data_sparsifier/data_norm_sparsifier.py:
```diff
@@ -80,7 +80,7 @@ def __get_block_level_mask(self, data,
                                               sparse_block_shape=sparse_block_shape)
 
-        mask = mask.squeeze()[:height, :width].contiguous()  # remove padding and make contiguous
+        mask = mask.squeeze(0).squeeze(0)[:height, :width].contiguous()  # remove padding and make contiguous
```
Is this what you want? Calling squeeze(0) twice only works if the first dim is 1. Maybe you mean `.squeeze(1).squeeze(0)`?

Example:

1. For `mask.shape = 1 x 1 x A x B`, we have:
    - `mask.squeeze(0).squeeze(0)` produces `A x B` 
    - `mask.squeeze(1).squeeze(0)` produces `A x B` 
2. For `mask.shape = X x 1 x A x B`, we have:
    - `mask.squeeze(0).squeeze(0)` produces `X x 1 x A x B` 
    - `mask.squeeze(1).squeeze(0)` produces `X x A x B`

macandro96 replied:
Ideally, this works for 2D data. If the data is 1D then we make it unsqueeze and proceed the same way. The mask that we get after scatter folding is 4D, so we squeeze it two times to make it 2D back again.

Commit:
Update on "[ao][sparsity] Bug Fix: data norm sparsifier not working with 1D tensors/parameters"


Issue:
Previously, the L1/L2 norm data sparsifier was not supported with
1D tensors or parameters.

Fix:
If the tensor is 1D, then unsqueeze it to make it look 2D and
perform the rest as usual. Also, added some 1D tensor in the
unit test to test this issue.

Test Plan:
```python test/test_ao_sparsity.py TestNormDataSparsifiers```

[ghstack-poisoned]

macandro96 commented:
@pytorchbot merge -g
